### DISCUSSION TITLE
apt update & apt upgrade  vs update.packages()

### DISCUSSION THREAD

**Discussion by pietro-coretto** [eddelbuettel/r2u#66] [General]
I followed the guidelines and set up a nice and clean install with `r2u` repos and `bspm` package.  

I installed both from shell with 
```
apt install r-cran-pkgname
```
and from a `sudo` session of `R` using 
```
install.package("pkgname")
```
and all stuff is installed under `/usr/lib/R/site-library` (except bspm) as expected.  This is wonderful, thanks!

My question is about updating packages.   Suppose that I installed `pkg_A` via `install.package("pkg_A")` under  admin privileges. When I do the next 
```
apt update && apt dist-upgrade
```
do I understand correctly that `pkg_A` now is updated/upgraded as if it was installed under apt?

**Comment by eddelbuettel** [comment]
Yes that works that way.  I have a few (long-running) machines (laptop, and ec2 instance) that operate that way. I only interface with apt _and my R installation updates as well_.

That is a _feature_ and not a bug. 

Of course, you _can_ still call `update.packages()` as well and because it will (as needed) call `install.packages()` which is traced thanks to `bspm` it will also hit the `apt` code path.  Still a feature.

/cc @enchufa2

**Reply by pietro-coretto** [reply]
You saved our poor lives.... 
Thank you infinitely !!!

**Reply by eddelbuettel** [reply]
Thanks for the very kind words but you know what they say about open source and an itch to scratch:  I had to save mine first :laughing:

**Comment by eddelbuettel** [comment]
See for example these two mastodon posts of mine (from a two-post thread) celebrating / illustrating this with screenshots:
- https://mastodon.social/@eddelbuettel/112464455366615093
- https://mastodon.social/@eddelbuettel/112464483600928649

**Comment by eddelbuettel** [comment]
And PS:

> and from a sudo session of R using

`bspm` allows you enable acquiting of `sudo` as needed so that you can do all this from a normal _user_ R session. See its docs, it is another option to set at startup like `bspm::enable()`.

**Comment by pietro-coretto** [comment]
Btw, I used your setup script  [add_cranapt_noble.sh](https://github.com/eddelbuettel/r2u/blob/master/inst/scripts/add_cranapt_noble.sh) and It worked perfect. 

Exploring the script I notice you start install `r-base-core`.  What is it exactly?  Does it contain `r-base` and `r-base-dev` in one package?

**Reply by eddelbuettel** [reply]
Do `apt-cache show` on the three and look carefully.  `r-base` is the (virtual) meta package enveloping all, `r-base-core` is the minimal "core" package needed to run (hence a valid choice where I needed it) and `-dev` is the development package.  We are topic drifting now so maybe open a new issue (or 'discussion' -- why are were anyway? Anyway....) to not dilute this subject's thread.
